packages/client-proxy/internal/proxy/proxy.go

```diff
@@ -80,22 +77,29 @@ func NewClientProxy(meterProvider metric.MeterProvider, serviceName string, port
 				return nil, reverseproxy.NewErrSandboxNotFound(sandboxId)
 			}
 
-			url := &url.URL{
+			// During migration period orchestrator proxy port is not set in some sandboxes
+			// https://linear.app/e2b/issue/ENG-3398
+			proxyPort := entry.OrchestratorProxyPort
+			if proxyPort == 0 {
+				proxyPort = consts.OrchestratorProxyPort
+			}
+
+			proxyUrl := &url.URL{
 				Scheme: "http",
-				Host:   fmt.Sprintf("%s:%d", nodeIP, orchestratorProxyPort),
+				Host:   fmt.Sprintf("%s:%d", entry.OrchestratorIP, proxyPort),
```

Comment:
Client proxy is now ready to proxy traffic to the orch running on non-standard ports.